### PR TITLE
886 Print network profiles

### LINKED ISSUE #886: Make printing available network configuration as a table to use during testing
**Author the user story for this feature**
As a User, I want to print available network configuration so I can see which network configuration I could use during testing.

**Is your feature request related to a problem? Please describe.**
Originally this was a User feature request to show available devices #835, but we would like to do more

**Describe the solution you'd like**
Printing available network-profiles are already done, but please print it as table.
Add prinitng network profiles to `test-environment`

**Additional context**
Ideally, in Flank we should have commands to list all types of testing information same as `gcloud`:
- ~~devices~~  #835 
- ~~available os software versions~~  #835 
- locales #884
- orientation #884
- network configuration #886
- provided software #885

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #886 

Command 

```

flank firebase test network-profiles list

```
and

```

flank network-profiles list

```

Should print table

```

┌────────────┬──────┬────────┬────────────┬───────────────────┬───────────┬───────┐
│ PROFILE_ID │ RULE │ DELAY  │ LOSS_RATIO │ DUPLICATION_RATIO │ BANDWIDTH │ BURST │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.040s │   0.001    │                   │  16000.0  │       │
│    LTE     │ down │ 0.040s │   0.001    │                   │  16000.0  │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.060s │   0.0015   │                   │  4000.0   │       │
│  LTE-poor  │ down │ 0.060s │   0.0015   │                   │  4000.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.060s │   0.001    │                   │  2000.0   │       │
│    HSPA    │ down │ 0.060s │   0.001    │                   │  4000.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.100s │   0.0015   │                   │   500.0   │       │
│ HSPA-poor  │ down │ 0.100s │   0.0015   │                   │  1000.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.060s │    0.02    │                   │  2000.0   │       │
│    UMTS    │ down │ 0.060s │    0.02    │                   │  2000.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.090s │    0.04    │                   │   384.0   │       │
│ UMTS-poor  │ down │ 0.090s │    0.04    │                   │   384.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.150s │    0.04    │                   │   384.0   │       │
│    EDGE    │ down │ 0.150s │    0.04    │                   │   384.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.200s │    0.08    │                   │   96.0    │       │
│ EDGE-poor  │ down │ 0.200s │    0.08    │                   │   96.0    │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.300s │    0.08    │                   │   172.0   │       │
│    GPRS    │ down │ 0.300s │    0.08    │                   │   172.0   │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.500s │    0.1     │                   │   48.0    │       │
│ GPRS-poor  │ down │ 0.500s │    0.1     │                   │   48.0    │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.500s │    0.08    │                   │    9.6    │       │
│    GSM     │ down │ 0.500s │    0.08    │                   │    9.6    │       │
├────────────┼──────┼────────┼────────────┼───────────────────┼───────────┼───────┤
│            │  up  │ 0.600s │    0.1     │                   │    2.4    │       │
│  GSM-poor  │ down │ 0.600s │    0.1     │                   │    2.4    │       │
└────────────┴──────┴────────┴────────────┴───────────────────┴───────────┴───────┘

```

also

```

flank firebase test android|ios test-environment 

```
should print this table at the end of the output. 

## WDYT

I added row separator to all tables to keep one style. In my opinion, it looks fine but on the other hand, the table with row separator takes more space. Tell me please what you think. Maybe we shouldn't add row separators to all tables.

## Checklist

- [X] Unit tested
- [X] release_notes.md updated
- [X] Printing network profiles on `flank firebase test network-profiles list`
- [X] Printing network profiles on `flank firebase test android|ios test-environment `

### CODE REVIEW

File: test_runner/src/test/kotlin/ftl/cli/firebase/test/networkprofiles/NetworkProfilesListCommandTest.kt [hunk 1]
```diff
@@ -33,19 +32,18 @@ class NetworkProfilesListCommandTest {
...
+                id = "test"
+                upRule = TrafficRule()
+                downRule = TrafficRule()
+            })
+            every {
+                GcTesting.networkConfiguration()
+            } returns configurationsMock
+            every { configurationsMock.asPrintableTable() } returns ""
+            CommandLine(NetworkProfilesListCommand()).execute()
+            verify { configurationsMock.asPrintableTable() }
```
Comment (piotradamczyk5) [inline test_runner/src/test/kotlin/ftl/cli/firebase/test/networkprofiles/NetworkProfilesListCommandTest.kt:46]: I think that we have response for this call-in mocked server. Could we use it somehow?

Comment (adamfilipow92) [reply test_runner/src/test/kotlin/ftl/cli/firebase/test/networkprofiles/NetworkProfilesListCommandTest.kt:46]: Yes, we have calls to TestEnvironmentCatalog I change the test to using a mock server. Thanks!

**Comment by piotradamczyk5** [comment]
Could you add also the possibility to run table using a command
`flank network-profiles list`
?

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidTestEnvironmentCommand.kt [hunk 2]
```diff
@@ -24,6 +26,7 @@ class AndroidTestEnvironmentCommand : Runnable {
         println(devicesCatalogAsTable(AndroidArgs.load(Paths.get(configPath)).project))
         println(supportedVersionsAsTable(AndroidArgs.load(Paths.get(configPath)).project))
         println(providedSoftwareAsTable())
+        println(GcTesting.networkConfiguration().asPrintableTable())
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidTestEnvironmentCommand.kt:29]: Could you update description to :
```
    header = ["Print available devices, OS versions, provided software list and network configuration to test against"],
    description = ["Print available Android devices, Android OS versions list, provided software and network configuration to test against"],
```

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosTestEnvironmentCommand.kt [hunk 3]
```diff
@@ -24,6 +26,7 @@ class IosTestEnvironmentCommand : Runnable {
         println(devicesCatalogAsTable(IosArgs.load(Paths.get(configPath)).project))
         println(softwareVersionsAsTable(IosArgs.load(Paths.get(configPath)).project))
         println(providedSoftwareAsTable())
+        println(GcTesting.networkConfiguration().asPrintableTable())
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosTestEnvironmentCommand.kt:29]: Could you update description to :
```
    header = ["Print available devices, OS versions, provided software list and network configuration to test against"],
    description = ["Print available iOS devices, iOS OS versions list, provided software and network configuration to test against"],
```
?

**Comment by piotradamczyk5** [comment]
```
WDYT
I added row separator to all tables to keep one style. In my opinion, it looks fine but on the other hand, the table with row separator takes more space. Tell me please what you think. Maybe we shouldn't add row separators to all tables.
```

For me defaults separator should not be printed cause tables become larger (readability is a plus, but the size is a minus :( )

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosTestEnvironmentCommand.kt (hunk 3 above)
Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosTestEnvironmentCommand.kt]: Changed

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidTestEnvironmentCommand.kt (hunk 2 above)
Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidTestEnvironmentCommand.kt]: Changed

**Comment by adamfilipow92** [comment]
> ```
> WDYT
> I added row separator to all tables to keep one style. In my opinion, it looks fine but on the other hand, the table with row separator takes more space. Tell me please what you think. Maybe we shouldn't add row separators to all tables.
> ```
> 
> For me defaults separator should not be printed cause tables become larger (readability is a plus, but the size is a minus :( )

So we thinking the same. I added a table style option and set default style without rows separation. Thanks for your feedback!

File: test_runner/src/test/kotlin/ftl/cli/firebase/test/networkprofiles/NetworkProfilesListCommandTest.kt (hunk 1 above)
Comment (adamfilipow92) [reply test_runner/src/test/kotlin/ftl/cli/firebase/test/networkprofiles/NetworkProfilesListCommandTest.kt]: I'm not 100% sure but probably I need to mock it to check printing is executed on command run.

**Comment by piotradamczyk5** [comment]
Great work 👍

**Pull request merged**